elyra/templates/kfp_template.jinja2

```diff
@@ -10,15 +10,15 @@ from kfp_notebook.pipeline import NotebookOp
 def create_pipeline():
     {% for key, operation in operations_list.items() %}
         {% set op_name = operation.notebook_name.split('.') %}
-        {% set operation_id = op_name[0] | replace ("/","_") %}
+        {% set operation_id = op_name[0] | replace ("/","_") | replace (" ","_") | replace ("-","_") %}
```

Comment:
Not sure we should go down this path to address this naming issue, as there can be many more characters that could cause failures. An safer approach would be to set `notebook_op_{{ operation_id }}` to a string like `notebook_op_N` (with `N` being an incremented counter, e.g. `notebook_op_1`, `notebook_op_2`, ...) would be my preferred approach.
By adding comments we can establish the relationship between the generic id and the source file, e.g.
```
# Operator for "lorem-isum notebook.ipynb"
notebook_op_1 = NotebookOp(name...
```

Reply:
udpated